R/aggregate_game_stats_def.R

```diff
@@ -0,0 +1,527 @@
+    dplyr::ungroup()
+
+  # Combine all stats -------------------------------------------------------
+
+  # combine all the stats together
+
+  player_df <- tackle_df %>%
+    dplyr::full_join(tackle_yards_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(pressure_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(int_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(safety_df, by = c("season","week","player_id", "team")) %>%
+    dplyr::full_join(fumble_df_own, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_df_opp, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_yds_own_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(fumble_yds_opp_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(penalty_df, by = c("season","week","player_id")) %>%
+    dplyr::full_join(touchdown_df, by = c("season","week","player_id")) %>%
+    dplyr::mutate_if(is.numeric, tidyr::replace_na, 0) %>%
+    dplyr::left_join(
+      nflreadr::load_players() %>%
+        dplyr::select(
+          "player_name" = "display_name",
+          "player_id" = "gsis_id",
+          "position"
+        ),
+      by = "player_id"
+    ) %>%
+    dplyr::select(tidyselect::any_of(c(
+
+      # game information
+      "season", "week",
+
+      # id information
+      "player_id", "player_name", "team", "recent_team", "position",
+
+      # tackle stats
+      "tackles", "tackles_solo", "tackles_with_assist", "tackle_assists",
+      "tackles_for_loss", "tackles_for_loss_yards" = "tfl_yards", "forced_fumbles",
+
+      # pressure stats
+      "sacks", "sack_yards", "qb_hit",
+
+      # coverage stats
+      "int", "pass_defended", "int_yards",
+
+      # misc stats
+      "td", "fumble", "fumble_recovery_own", "fumble_recovery_yards_own",
+      "fumble_recovery_opp", "fumble_recovery_yards_opp", "safety", "penalty"
+
+    ))) %>%
+    dplyr::filter(!is.na(.data$player_id), !is.na(.data$player_name)) %>%  ### player_name only because of Terell Bonds
+    dplyr::arrange(.data$player_id, .data$season, .data$week)
+
+  # if user doesn't want week-by-week input, aggregate the whole df
+  if (isFALSE(weekly)) {
+    player_df <- player_df %>%
+      dplyr::group_by(.data$player_id) %>%
+      dplyr::summarise(
+        player_name = custom_mode(.data$player_name),
+        games = dplyr::n(),
+        recent_team = dplyr::last(.data$team),
+        position = custom_mode(.data$position),
+        tackles = sum(.data$tackles),
+        tackles_solo = sum(.data$tackles_solo),
+        tackles_with_assist = sum(.data$tackles_with_assist),
+        tackle_assists = sum(.data$tackle_assists),
+        tackles_for_loss = sum(.data$tackles_for_loss),
+        tackles_for_loss_yards = sum(.data$tackles_for_loss_yards),
+        forced_fumbles = sum(.data$forced_fumbles),
+        sacks = sum(.data$sacks),
+        sack_yards = sum(.data$sack_yards),
```

Comment:
Is there an official source that provides sack yards we should be aligning to? If desired this can be done as `sum(.data$sacks * .data$sack_yards) * -1` but I would make sure we're matching NFL.com's sack yards here if they measure it.

Reply:
Yes, Sack yards on defense is an official NFL GSIS stat (id 83, it seems). http://www.nflgsis.com/gsis/documentation/Partners/StatIDs.html

I didn't see stat yards on NFL.com, but everyone I know who tallies them (most notably ESPN at http://www.nflgsis.com/gsis/documentation/Partners/StatIDs.html and Sportradar, which I see via MFL, all track stat yards in the positive and in this split manner).

I think the formula you propose is the way to go. Let's do it!

Reply:
Done in upcoming commit